test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -1163,120 +1302,68 @@ def _test_ddp_zero_overlap(
                     self.assertEqual(p1, p2)
 
                 # Check that the parameters were updated
-                self.assertNotEqual(init_params_overlap, list(ddp_model_overlap.parameters()))
+                self.assertNotEqual(
+                    init_params_overlap, list(ddp_model_overlap.parameters()),
+                )
 
                 # Ensure that this test runs independently
                 dist.barrier()
 
+    # NOTE: The test is skipped if using Windows since functional optimizers
+    # are not currently supported.
     @common_distributed.skip_if_win32()
     @common_distributed.requires_nccl()
     @common_distributed.skip_if_no_gpu
     @common_distributed.skip_if_rocm
-    def test_ddp_with_zero_step_parity_gpu(self):
-        r"""
-        Check that overlapping DDP with ZeRO using ``hook_with_zero_step()``
-        achieves parity with DDP using a local optimizer when running on GPU.
-
-        NOTE: The test is skipped if using Windows since functional optimizers
-        are not currently supported.
+    @parametrize(
+        "use_gpu",
+        [True],
+        # Add `False` once the Gloo sync issue causing hangs is fixed
+        # See: https://github.com/pytorch/pytorch/issues/62300
```

Comment:
wondering if we have plans to fix this issue in gloo? 

cc @awgu @mrshenli @cbalioglu

Reply:
Assigned #62300 to myself. I plan to work on it as part of the overhaul of our process group APIs.